test-framework/src/main/kotlin/com/epam/drill/e2e/E2ETest.kt

```diff
@@ -68,66 +151,36 @@ abstract class E2ETest : AdminTest() {
                     val globLaunch = application.launch(handler) {
                         watcher?.invoke(asyncEngine, glob)
                     }
-                    val cs = mutableMapOf<String, AdminUiChannels>()
+
+                    val cs = ConcurrentHashMap<String, AdminUiChannels>()
                     coroutineScope {
-                        agents.map { (_, xx) ->
-                            val (ag, startClb, connect, thens) = xx
-                            val ui = AdminUiChannels()
-                            cs[ag.id] = ui
+                        agents.map { (_, queue) ->
                             launch(handler) {
-                                startClb()
-                                val uiE = UIEVENTLOOP(cs, uiStreamDebug, glob)
-                                with(uiE) { application.queued(appConfig.wsTopic, uiIncoming) }
-
-                                //create the '/agent/attach' websocket connection
-                                ut.send(uiMessage(Subscribe("/agents/${ag.id}")))
-                                ut.send(uiMessage(Subscribe("/agents/${ag.id}/builds")))
-
-                                ui.getAgent()
-                                ui.getBuilds()
-                                delay(50)
-
-                                handleWebSocketConversation(
-                                    "/agent/attach",
-                                    wsRequestRequiredParams(ag)
-                                ) { inp, out ->
-                                    glob.receive()
-                                    val apply = Agent(application, ag.id, inp, out, agentStreamDebug).apply { queued() }
-                                    connect(
-                                        asyncEngine,
-                                        ui,
-                                        apply
-                                    )
-                                    while (globLaunch.isActive)
-                                        delay(100)
-
-                                }
-                                thens.forEach { (ain, it) ->
-                                    glob.receive()
+                                val currentInstance = queue.poll()
+                                val agentId = currentInstance.first.id
+                                if (cs[agentId] == null) {
```

Comment:
```suggestion
                                cs[agentId]?.let {
```

Reply:
cs[agentId]?.let -> if (cs[agentId] != null) 
but we need this block whent cs[agentId] == null